extensions/indexes/lucene/src/main/java/org/exist/indexing/lucene/LuceneFieldConfig.java

```diff
@@ -169,7 +185,12 @@ protected void processResult(Sequence result, Document luceneDoc) throws XPathEx
 
     @Override
     protected void processText(CharSequence text, Document luceneDoc) {
-        final Field field = convertToField(text.toString());
+        final Field field;
```

Comment:
final Field field = binary ? convertToDocValue(text) : convertToField(text);

would be acceptable for me :-)